mlflow/tracking/client.py

```diff
@@ -1709,6 +1727,54 @@ def log_table(
             # for data like {"inputs": "What is MLflow?"}
             except ValueError:
                 data = pd.DataFrame([data])
+
+        # Check if the column is a `PIL.Image.Image` or `mlflow.Image` object
+        # and save filepath
+        if len(data.select_dtypes(include=["object"]).columns) > 0:
+            try:
+                from PIL import Image
```

Comment:
(I thought I have commented this but didn't remember your reply..)

This block could cause unnecessary exception for people having object not Image in the column. IMO we can avoid importing PIL and use builtin api like `__class__` or so to check if that's a PIL type. For users willing to log images with `log_table`, they will hit error at other places if missing PIL.

Reply:
oh sorry, for this I was going to say that its gated only when we see an "object" type - so only if a user logs an object then we run the check. It wouldn't make too much sense to silently fail here or not store the image properly.

However, it makes sense that a less restrictive check can be used so that users will have a better UX here. I've updated the PR to check for specific methods that occur in the PIL image class.